internal/tools/kops/cluster.go

```diff
@@ -51,6 +67,37 @@ func (c *Cmd) UpdateCluster(name string) error {
 	return nil
 }
 
+// UpgradeCluster invokes kops upgrade cluster, using the context of the created Cmd.
+func (c *Cmd) UpgradeCluster(name string) error {
```

Comment:
Confusingly, the kops documentation [suggests](https://github.com/kubernetes/kops/blob/master/docs/upgrade.md#other-notes):
> In general, we recommend that you upgrade your cluster one minor release at a time (1.7 --> 1.8 --> 1.9). Although jumping minor versions may work if you have not enabled alpha features, you run a greater risk of running into problems due to version deprecation.

Instead of leveraging their automated update, should we instead consider wrapping the manual update steps and accept a target version instead?

Reply:
Yeah, so I read the same thing in the kops documentation and had similar thoughts.

I am approaching this uncertainty with the following plan:

We probably want to have the `kops upgrade` command wrapped as it is in my PR, even if we don't use it much in the longterm. I would be curious to use it as we test out a full Mattermost provisioning and upgrade cycle just to see what we experience. In the longterm, once we have some experience, we may want to move to the stepping process you outlined.

i.e. Let's use `kops upgrade` for now so we can make sure that it works as expected, even if we don't want to use it for cluster upgrades in the future. How does that sound?

Reply:
Agreed re: incremental steps. Can you file a ticket to capture the need to step through the upgrades one-by-one?